x/authenticator/authenticator/cosmwasm.go

```diff
@@ -270,11 +107,20 @@ func (cwa CosmwasmAuthenticator) Track(ctx sdk.Context, account sdk.AccAddress,
 		return errorsmod.Wrap(sdkerrors.ErrInvalidRequest, "failed to encode msg")
 	}
 
+	jsonMsg, err := json.Marshal(msg)
```

Comment:
mostly to make it easier on contract authors. But added it when applying the refactor to message filter since it has to serialize the message to json and wanted to avoid having to go from any to sdk.msg to json. 

Happy to remove it if it's not helpful.

Reply:
Then the `Any` problem will occur for msg filter too? But for cosmwasm I think just proto should be fine:

- `osmosis-std` can handle that (it has cosmos sdk msgs now and it's proto capability is more robust than json since it does not have field naming mismatch & some other serialization problem which I have to patch every now and then)
- even if there is no generated type in case other chain adopt it, `prost` will work just fine (or maybe just [AnyBuf](https://github.com/noislabs/anybuf) if they wanna go light weight)
- missing json conditionally is quite an unexpected api IMO, no obvious error shown

Reply:
> Then the Any problem will occur for msg filter too? 

yeah, but we can deal with it there. I think it's a bit more complex in the message filter as we will already have to get the msg from the any.

But yes, let's remove the json!